rfd/0036/user-stories/jupiter-example-com-jenkins.md

```diff
@@ -0,0 +1,77 @@
+
+### GitHub
+
+The code is hosted in GitHub. Most everything here varies only in detail with GitLab or Bitbucket. Each project (ex. jupiter.example.com) has a GitHub repository. The GitHub repository is configured with a webhook that sends push events to Jenkins. These push events include information on what branch and commit was pushed.
+
+### Jenkins jobs
+
+Jenkins should be configured with two jobs, one for development and one for production. The separate production job allows for manual approval of the build while reducing risk of accidental deployment (see below).
+
+### Jenkins development/test job
+
+This Jenkins job is configured to receive GitHub webhook pushes. The Jenkins job should run build and deployment tasks defined in the code repository so that the process for deployment is owned by the team that owns the code.
+
+Each GitHub branch has an associated Mariposa project. The Jenkins job clones a template manifest (from the repo) for each new branch. Each changeset received on a given branch will be deployed as a new version of that manifest.
+
+The job task might look like the following:
+
+- Run `docker-compose build` locally
+- Tags containers with the `GIT_BRANCH` and `GIT_COMMIT`
+- Containers to a registry (optionally a private registry)
+- Render manifest template to manifest
+- `mariposa push manifest`
+- `mariposa deploy version`
+
+Mariposa will then see a new version of the manifest and deploy those containers for testing.
+
+> Open question: when do you shut down containers or delete a project associated with a branch? We can receive `delete` events from GitHub web hooks, but it’s not clear Jenkins jobs can do anything with them.
+
+
+### Jenkins production job
+
+Despite the automation for continuous build and deploy of the staging projects, the deploys to the live site require manual promotion. After reviewing the relevant staging site and any logs generated by Jenkins, the operators merge a PR to master, triggering a new build of master that is tagged with the date and time of the build (the builds collect resources beyond the scope of the git repo, so the git SHA is not a consistent identifier). The master branch is built and staged using the same logic as the development jobs above in a project named <repo-name>-branch-master.
+
+After testing and previewing that final staging build, the operators can choose to manually deploy that build to the live site. This manual deployment is done via a separate production job. This Jenkins job is not configured to receive GitHub webhook pushes. Instead, it is a parameterized job which deploys the parameterized tag (with a default like “production”). The end-user tags the release in git with a version and the tag.
+
+```
+git tag v1.2.3
+git tag production
```

Comment:
Please discuss in https://github.com/misterbisson/rfd/pull/3#discussion_r97371349